### PR TITLE
Cherry-pick: Add driverJars parameter to JdbcIO. (#25824)

### PR DESCRIPTION
**Pull request by bvolpato**
Cherry-picking https://github.com/apache/beam/pull/25824 to the release.

### CODE REVIEW

**Comment by bvolpato** [comment]
R: @jrmccluskey

**Comment by jrmccluskey** [comment]
Run Python_Integration PreCommit

**Comment by Abacn** [comment]
Possibly related: https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Python311/105/

**Comment by Abacn** [comment]
Run Python 3.11 PostCommit

File: sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java [hunk 1]
```diff
@@ -67,6 +67,8 @@ public Schema configurationSchema() {
         .addNullableField("fetchSize", FieldType.INT16)
         .addNullableField("outputParallelization", FieldType.BOOLEAN)
         .addNullableField("autosharding", FieldType.BOOLEAN)
+        .addNullableField("maxConnections", FieldType.INT16)
```
Comment (Abacn) [inline sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java:70]: This is a breaking change for xlang schema transform, see #25062

in general we may need a better way to make this kind of change not breaking xlang

**Review by Abacn — CHANGES_REQUESTED** [review]

File: sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java [hunk 2]
```diff
@@ -202,11 +204,14 @@ protected JdbcIO.DataSourceConfiguration getDataSourceConfiguration() {
         dataSourceConfiguration = dataSourceConfiguration.withConnectionInitSqls(initSqls);
       }
 
-      if (config.getSchema().hasField("maxConnections")) {
-        @Nullable Integer maxConnections = config.getInt32("maxConnections");
-        if (maxConnections != null) {
-          dataSourceConfiguration = dataSourceConfiguration.withMaxConnections(maxConnections);
-        }
+      @Nullable Integer maxConnections = config.getInt32("maxConnections");
```
Comment (Abacn) [inline sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java:207]: I see, the actual breaking change happens here. This if clause is removed

File: sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java (hunk 1 above)
Comment (Abacn) [reply sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java:70]: I see, I think I find the cause. Here maxConnections is "FieldType.INT16" but in jdbc.py it is "typing.Optional[int]". This configuration gets encoded by ShortCoder then decode with VarIntCoder and data corruption happens.

Comment (Abacn) [reply sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java:70]: opened #26480

**Comment by bvolpato** [comment]
Updated with patch provided by @Abacn at https://github.com/apache/beam/pull/25062. Thanks Yi.

**Comment by Abacn** [comment]
Run Python 3.11 PostCommit

**Review by Abacn — APPROVED** [review]
Thanks, LGTM

**Comment by jrmccluskey** [comment]
Run Python 3.11 PostCommit

**Comment by jrmccluskey** [comment]
Run Python_Integration PreCommit

**Comment by jrmccluskey** [comment]
I'm not anticipating this PR significantly impacting the Python tests I'm re-running, will merge after those have any result

**Comment by Abacn** [comment]
Dataflow tests are flaky recently due to no worker available in us-central1 region.

**Comment by jrmccluskey** [comment]
That would explain the flakey failures I've been seeing on streaming integration tests. Thank you for the context!

**Pull request merged**